### ISSUE TITLE
class not found 

### ISSUE THREAD

**Issue by skuuub** [microsoft/vscode-java-debug#1335]
Type: <b>Performance Issue</b>

when I try to run a new Java program, it gives me this error " Error: Could not find or load main class text
Caused by: java.lang.ClassNotFoundException: text ".
It won't work and if I try to run an old program it works but if I change a single line of code and undo it, the whole program won't work and spits out the same error

Extension version: 0.50.0
VS Code version: Code 1.77.3 (Universal) (704ed70d4fd1c6bd6342c436f1ede30d1cff4710, 2023-04-12T09:19:37.325Z)
OS version: Darwin arm64 22.4.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 6, 5|
|Memory (System)|8.00GB (0.13GB free)|
|Process Argv|--crash-reporter-id 8eb60dd9-d33a-4b65-8605-60a55f6a96d1|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
   15	   197	  4106	code main
    0	    82	  4111	   gpu-process
    0	    33	  4113	   utility-network-service
    0	   344	  4114	window [1] (OOP.java — VSjava)
    0	   139	  4125	shared-process
    0	    66	  4129	     ptyHost
    0	     0	  4198	       /bin/zsh -il
    0	     0	  4277	       /bin/zsh -l
    0	     0	  4359	     /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	    57	  4126	fileWatcher [1]
    0	   385	  4127	extensionHost [1]
    0	   344	  4177	     /Users/oluwabukunmijaiyeola/.vscode/extensions/redhat.java-1.18.0-darwin-arm64/jre/17.0.7-macosx-aarch64/bin/java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:disable -javaagent:/Users/oluwabukunmijaiyeola/.vscode/extensions/redhat.java-1.18.0-darwin-arm64/lombok/lombok-1.18.27.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/Users/oluwabukunmijaiyeola/Library/Application Support/Code/User/workspaceStorage/2bbc24249580735ec9764ad01b7a4903/redhat.java -Daether.dependencyCollector.impl=bf -jar /Users/oluwabukunmijaiyeola/.vscode/extensions/redhat.java-1.18.0-darwin-arm64/server/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration /Users/oluwabukunmijaiyeola/Library/Application Support/Code/User/globalStorage/redhat.java/1.18.0/config_mac -data /Users/oluwabukunmijaiyeola/Library/Application Support/Code/User/workspaceStorage/2bbc24249580735ec9764ad01b7a4903/redhat.java/jdt_ws
    0	   180	  4192	     electron-nodejs (/private/var/folders/x3/njdbgzc5759bqszdfmynlnsw0000gn/T/AppTranslocation/62FA25BD-3AEE-4796-9FB9-CF9A2B1F7A41/d/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) --ms-enable-electron-run-as-node /Users/oluwabukunmijaiyeola/.vscode/extensions/streetsidesoftware.code-spell-checker-2.20.4/packages/_server/dist/main.js --node-ipc --clientProcessId=4127)
    0	    57	  4246	     electron-nodejs (/private/var/folders/x3/njdbgzc5759bqszdfmynlnsw0000gn/T/AppTranslocation/62FA25BD-3AEE-4796-9FB9-CF9A2B1F7A41/d/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) --ms-enable-electron-run-as-node /private/var/folders/x3/njdbgzc5759bqszdfmynlnsw0000gn/T/AppTranslocation/62FA25BD-3AEE-4796-9FB9-CF9A2B1F7A41/d/Visual Studio Code.app/Contents/Resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=4127)
    6	    90	  4355	window [2] (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (OOP.java — VSjava)
|    Folder (VSjava): 6 files
|      File types: java(4) json(1)
|      Conf files: launch.json(1)
|      Launch Configs: java(6);
```

</details>
<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
nodejswelcome1:30587005
3biah626:30602489
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
showlangstatbar:30672706
vsccsb:30705552
azdwalk:30721579
pythonms35:30701012
pythonfmttextcf:30716743

```

</details>

**Comment by testforstephen** [comment]
could you pls provide a sample project to help reproduce the issue?

**Comment by skuuub** [comment]
> could you pls provide a sample project to help reproduce the issue?

I already solved the issue after I tried 3 different things. I first uninstalled all the Java extensions I installed, then I uninstalled Vs code, then I updated my JDK version which was one version behind. after O did those, I re-installed vs and the Java extensions, and then the issue went away. I should have kept track but I was in a hurry. I personally thing it either had something to do with the JDK or the extensions.
